fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java

```diff
@@ -0,0 +1,338 @@
+            for (SelectListItem selectListItem : selectItems) {
+                row.add(selectListItem.getExpr());
+
+                String name;
+                if (selectListItem.getAlias() != null) {
+                    name = selectListItem.getAlias();
+                } else if (selectListItem.getExpr() instanceof SlotRef) {
+                    name = ((SlotRef) selectListItem.getExpr()).getColumnName();
+                } else {
+                    name = selectListItem.getExpr().toColumnLabel();
+                }
+                columnNames.add(name);
+            }
+            List<ArrayList<Expr>> rows = new ArrayList<>();
+            rows.add(row);
+            return new ValuesRelation(rows, columnNames);
+        }
+
+        boolean isDistinct = context.setQuantifier() != null && context.setQuantifier().DISTINCT() != null;
+        return new SelectRelation(
+                new SelectList(selectItems, isDistinct),
+                from,
+                visitIfPresent(context.where, Expr.class).orElse(null),
+                visitIfPresent(context.groupingElement(), GroupByClause.class).orElse(null),
+                visitIfPresent(context.having, Expr.class).orElse(null));
+    }
+
+    // ------------------------------------------- Relation -------------------------------------------
+
+    @Override
+    public ParseNode visitTableName(StarRocksParser.TableNameContext context) {
+        QualifiedName qualifiedName = getQualifiedName(context.qualifiedName());
+        TableName tableName = qualifiedNameToTableName(qualifiedName);
+        TableRelation tableRelation = new TableRelation(tableName);
+        if (context.hint() != null) {
+            for (TerminalNode hint : context.hint().IDENTIFIER()) {
+                if (hint.getText().equalsIgnoreCase("_META_")) {
+                    tableRelation.setMetaQuery(true);
+                }
+            }
+        }
+        return tableRelation;
+    }
+
+    @Override
+    public ParseNode visitAliasedRelation(StarRocksParser.AliasedRelationContext context) {
+        Relation child = (Relation) visit(context.relationPrimary());
+
+        if (context.identifier() == null) {
+            return child;
+        }
+        Identifier identifier = (Identifier) visit(context.identifier());
+        child.setAlias(new TableName(null, identifier.getValue()));
+        return child;
+    }
+
+    @Override
+    public ParseNode visitJoinRelation(StarRocksParser.JoinRelationContext context) {
+        Relation left = (Relation) visit(context.left);
+        Relation right;
+
+        if (context.CROSS() != null) {
+            right = (Relation) visit(context.right);
+            return new JoinRelation(JoinOperator.CROSS_JOIN, left, right, null, false);
+        }
+
+        right = (Relation) visit(context.rightRelation);
+
+        JoinOperator joinType = JoinOperator.INNER_JOIN;
+        if (context.joinType().LEFT() != null) {
```

Comment:
Could these if-else be replaced with a for loop?

Reply:
how replaced with a for loop?

Reply:
Maybe, assemble all join type into a list, and then iterate the list and find the first not-null value?

Reply:
What is the purpose of doing this? Is there any benefit or advantage over if?

Reply:
Well, just ignore this suggestion. I prefer for-loop to if-else.